typescript/sdk/src/deploy/core/HyperlaneCoreDeployer.ts

```diff
@@ -59,18 +64,54 @@ export class HyperlaneCoreDeployer extends HyperlaneDeployer<
   async deployInterchainGasPaymaster(
     chain: ChainName,
     proxyAdmin: ProxyAdmin,
+    gasOracleContracts: GasOracleContracts,
     deployOpts?: DeployOptions,
   ): Promise<
     ProxiedContract<InterchainGasPaymaster, TransparentProxyAddresses>
   > {
-    return this.deployProxiedContract(
+    const beneficiary = this.configMap[chain].igp.beneficiary;
+    const igp = await this.deployProxiedContract(
       chain,
       'interchainGasPaymaster',
-      [],
+      [beneficiary],
       proxyAdmin,
-      [],
+      [beneficiary],
```

Comment:
why does this appear twice?

Reply:
once for the constructor and once for the initializer

Kind of related to the convo we had in person where we're doing something a bit weird by calling the initializer in the constructor